### PR TITLE
New Relic documentation

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Added New Relic docs.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

## Notion ticket link

> [New Relic documentation](https://www.notion.so/rudderstacks/New-Relic-destination-8268ba8babc044cc98628a8fcb19caa8)

### CODE REVIEW

File: destinations/analytics/new-relic.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+- **Data Center**: Choose your data center associated with your New Relic account. By default, it is set to **US**.
+- **Custom Default Event Type**: Use this field to define any custom `eventType` for any unmapped `track` calls. By default, RudderStack sets the `eventType` as `rudderstack`.
+- **Send Device Context**: If this setting is enabled, RudderStack flattens all the contextual fields delimited by a period (`.`) before sending them to New Relic.
+
+<div class="successBlock">
+
+Enabling this setting is recommended as getting rich device context is very useful in New Relic.
+</div>
+- **Send User ID and Anonymous ID**: Enable this option to send `userId` and `anonymousId` in your events to New Relic.
```
Comment (ameypv-rudder) [inline destinations/analytics/new-relic.mdx:59]:
```suggestion

- **Send User ID and Anonymous ID**: Enable this option to send `userId` and `anonymousId` in your events to New Relic.
```

File: destinations/analytics/new-relic.mdx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    "subject": "Resume validation",
+    "sendtime": "2022-01-01",
+    "sendlocation": "alex@example.com"
+  }, {
+    traits: {
+      email: "alex@example.com"
+    }
+  }
+);
+```
```
Comment (ItsSudip) [inline destinations/analytics/new-relic.mdx:82]:
```suggestion
module.exports = { process, processRouterDest };
rudderanalytics.track(
  "Email Opened",
  {
    subject: "Resume validation",
    sendtime: "2022-01-01",
    sendlocation: "alex@example.com"
  },
  {
    context: {
      traits: {
        email: "alex@example.com"
      }
    }
  }
);

```

Comment (ameypv-rudder) [reply destinations/analytics/new-relic.mdx:82]: @ItsSudip is the following line required?

```
module.exports = { process, processRouterDest };
```

File: destinations/analytics/new-relic.mdx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
```
Comment (ItsSudip) [inline destinations/analytics/new-relic.mdx:83]: We have to add a few lines here.

```
We are applying the following rules to the event properties:

- booleans will be transformed into strings
- [reserved](https://docs.newrelic.com/docs/data-apis/custom-data/custom-events/data-requirements-limits-custom-event-data/#:~:text=to%20the%20logs.-,NRQL%20syntax%20terms,-If%20you%20need) nrql words are wrapped with backticks
- [reserved](https://docs.newrelic.com/docs/data-apis/custom-data/custom-events/data-requirements-limits-custom-event-data/#reserved-words) words will be removed removed
- dates will be converted to ISO strings- arrays and objects are removed
```

**Comment by ameypv-rudder** [comment]
There were some issues in rebasing this branch with the newly structured `main` branch. Hence opened a new PR along with the changes suggested by @ItsSudip. Closing this one, as a result.

**Pull request closed without merging**